src/nvim/drawline.c

```diff
@@ -1420,18 +1377,46 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, int mod_top, bo
     line_attr_lowprio_save = wlv.line_attr_lowprio;
   }
 
-  line = end_fill ? "" : ml_get_buf(wp->w_buffer, lnum, false);
-  ptr = line;
+  if (spv->spv_has_spell && !number_only) {
```

Comment:
`number_only`, `end_fill` and `has_fold`, are used quite inconsistently throughout `win_line()`. I think a lot of times if one is used to guard a block, the others should be as well. That is not always the case currently.

Might verify that this doesn't break anything and/or introduce a new boolean to skip these blocks later...

Reply:
In other places this seems already guarded by other things like `v > 0`.

Reply:
Why not just add `&& !has_fold && !end_fill` to the spell check block?

Reply:
It matches the old code better and make it less necessary to decide whether to add `&& !has_fold && !end_fill` in other places where `spv_has_spell` appears.